connector/src/main/java/org/geysermc/connector/network/session/GeyserSession.java

```diff
@@ -479,6 +485,37 @@ public void packetError(PacketErrorEvent event) {
         }).start();
     }
 
+    public void handleDownstreamPacket(Packet packet) {
+        //handle consecutive respawn packets
+        if (packet.getClass().equals(ServerRespawnPacket.class)) {
+            manyDimPackets = lastDimPacket != null;
+            lastDimPacket = (ServerRespawnPacket) packet;
+            return;
+        } else if (lastDimPacket != null) {
+            EventResult<DownstreamPacketReceiveEvent<?>> result = EventManager.getInstance().triggerEvent(DownstreamPacketReceiveEvent.of(this, lastDimPacket));
+            if (!result.isCancelled()) {
+                PacketTranslatorRegistry.JAVA_TRANSLATOR.translate(result.getEvent().getPacket().getClass(), result.getEvent().getPacket(), this);
+            }
+            lastDimPacket = null;
+        }
+
+        // Required, or else Floodgate players break with Bukkit chunk caching
+        if (packet instanceof LoginSuccessPacket) {
```

Comment:
This is the code that calls the translators.

Reply:
I'm referring to the LoginSuccessPacket specifically, which is calling logic that could be a packet translator class.

Reply:
Yeah agreed. This applies to most the stuff in Upstream handler as well so that we can reduce duplicate code. I held off changing that yet simply because maintaining too large a diff is hard to keep merging master in. However if this pr is likely to be mainline soon it's worth cleaning that up too.

Reply:
I've had a closer look at this and I don't think its worth changing yet. Once plugin support is merged I think floodgate can actually register as a plugin and use the plugin events itself instead of hooking into geyser mainline code which would allow this section to be removed.

What I was referring to in my comment was [UpstreamPacketHandler](https://github.com/GeyserMC/Geyser/blob/master/connector/src/main/java/org/geysermc/connector/network/UpstreamPacketHandler.java) which can probably have much of it moved to the individual translators instead however that is probably out of scope for this PR.